File: src/main/java/de/hinundhergestellt/jhuh/service/ready2order/SingleArtooMappedProduct.java
package de.hinundhergestellt.jhuh.service.ready2order;

import de.hinundhergestellt.jhuh.vendors.ready2order.ArtooProduct;

import java.util.List;

public class SingleArtooMappedProduct extends ArtooMappedProduct {

    private final ArtooProduct product;

    SingleArtooMappedProduct(ArtooProduct product) {
        super(List.of(new ArtooMappedVariation(product)));
        this.product = product;
    }

    @Override
    public String getName() {
        return product.getName();
    }
}

File: src/main/java/de/hinundhergestellt/jhuh/service/ready2order/SingleArtooMappedProduct.kt
package de.hinundhergestellt.jhuh.service.ready2order

import de.hinundhergestellt.jhuh.vendors.ready2order.ArtooProduct

class SingleArtooMappedProduct internal constructor(
    private val product: ArtooProduct
) : ArtooMappedProduct(listOf(ArtooMappedVariation(product))) {

    override val id = "single-${product.id}"
    override val name by product::name
    override val description by product::description
}
